### ISSUE TITLE
ZachFOL2019 checker accepts wrong EE inference

### ISSUE THREAD

**Issue by rzach** [Carnap/Carnap#37]
Line 11 is incorrect since the eigenvariable condition is violated:
![Screenshot from 2019-11-04 16-15-10](https://user-images.githubusercontent.com/3122348/68166054-5d3ea880-ff1e-11e9-8a9d-b26866dc4cec.png)
First problem of https://carnap.io/assignments/Calgary%20Phil%20279%20Fall%202019/279%20F19%20Problem%20Set%204.md
```
∀y(A(y) ∨ B(y)):PR
∀x(A(x) → C(x)):PR
∃x¬C(x):PR
A(a) ∨ B(a):AE 1
    -C(a):AS
    A(a) → C(a):AE 2
        A(a):AS
        C(a):->E6,7
        _|_:-E 5,8
    -A(a):-I 7-9
-A(a):EE 3,5-10
    A(a):AS
    _|_:-E 11,12
    B(a):X 13
--
    B(a):AS
    B(a):R16
B(a):|E 4, 12-14,16-17
∃zB(z):EI 18
```

**Comment by gleachkr** [comment]
I am pretty sure this is a bug introduced by merging in the Gentzen-Prawitz tree checking. It changes the way that rules are allowed to fall back on auxiluary inference schemes when their restrictions are violated, and I think what's happening is that because of the eigenvariable violation, CalgaryFOL is illegitimately falling back on a version of the rule that doesn't eliminate the assumption `-C(a)` (that's the reason for the warning triangle next to the sequent up top - undischarged assumption). If that's right, then this ought to be an easy fix, although I need to go over a couple of other systems to make sure that a similar issue hasn't arisen.

**Comment by gleachkr** [comment]
Closed by eba4b13.
